src/components/dialog/dialog.js

```diff
@@ -734,12 +734,37 @@ function MdDialogProvider($$interimElementProvider) {
           if (element === document.body) {
             return;
           }
-          var children = element.parentNode.children;
+          var children = element.parentNode.children,
+            attrToSet = 'md-saved-tabindex',
+            attrToDefault = 'tabindex',
+            tabModifierSelector = '[md-saved-tabindex]:not([md-saved-tabindex="-1"])';
+
+          if (isHidden) {
+            attrToSet = 'tabindex';
+            attrToDefault = 'md-saved-tabindex';
+            tabModifierSelector = 'button:not([tabindex="-1"]),\
+              a[href]:not([tabindex="-1"]),\
+              input:not([tabindex="-1"]),\
+              textarea:not([tabindex="-1"]),\
+              select:not([tabindex="-1"]),\
+              [tabindex]:not([tabindex="-1"])';
```

Comment:
Nicely done here, this ensures custom directives will get the same behavior as natively focusable elements.